flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/sink/KafkaWriter.java

```diff
@@ -376,6 +378,11 @@ private void initMetrics(FlinkKafkaInternalProducer<byte[], byte[]> producer) {
                         others,
                         kafkaSinkContext.getNumberOfParallelInstances())) {
             return log.getTransactionsToAbort();
+        } catch (KafkaException e) {
+            LOG.warn(
+                    "Cannot abort transactions before startup this might leave lingering transactions from previous executions.",
```

Comment:
I'd still give a pointer to the ACLs because it's not so obvious. But I'd also add the possibility of temporary Kafka problems as you have said. It's a log that is only emitted during recovery, so it can be more verbose. (although it will appear for each subtask)